src/runtime-rs/crates/hypervisor/README.md

```diff
@@ -89,6 +92,215 @@ pub trait Hypervisor: Send + Sync {
+VMM should have their own orientations.
+
+#### Explanations
+```rust
+async fn prepare_vm(&self, id: &str, netns: Option<String>) -> Result<()>
+```
+1. What is expected?
+* In general, `prepare_vm()` need the VMM to prepare everything **that VMM needs before boot**. 
+For example, maybe fill in some fields of VMM, or maybe create something on host that the VMM needs when booting. But only some preparations need to be done.
+
+
+2. What DBS does?
+* DBS fills in some of its fields, e.g. `id, state, jailer_root, vm_path, netns`, and create a Unix Domain Socket(uds) for communication. Note that the uds is not directly added to the vmm, but added to a buffer. The devices
+in the buffer will be added after the VM starts.
+
+---
+
+```rust
+async fn start_vm(&self, timeout: i32) -> Result<()>
+```
+1. What is expected?
+* In general, `start_vm()` need the VMM and kernel to start and function correctly, which means ALL other functions from the trait `Hypervisor` should be able to return an expected result. This may be based on the preparation that `prepare_vm()` has done.
+
+2. What DBS does?
+* DBS first run its VMM server.
+    * It creates the jailer path of the VM, setup the channel for thread communication, and start a thread to run its event_loop, since DBS runs as the thread within the process. Other VMM may start another process.
+* DBS then starts its kernel.
+    * It does normal things as a VMM, prepare kernel_params, initrd, image, rootfs, etc. and runs with its kernel started.
+
+---
+
+```rust
+async fn stop_vm(&self) -> Result<()>
+```
+1. What is expected?
+* In general, `stop_vm()` wants you to exit the VMM from its execution.
+
+2. What DBS does?
+* It sends an stopVM request to DBS VMM server, the VMM server will receive the request and exit its execution from the thread called `vmm_thread`.
+
+---
+
+```rust
+    async fn pause_vm(&self) -> Result<()>;
+```
+1. What is expected?
+* In general, `pause_vm()` wants you to pause the VMM from its execution like it said. The ways to actually pause the VMM differs from VMMs.
+
+2. What DBS does?
+* Not implemented in DBS.
+
+---
+
+```rust
+    async fn save_vm(&self) -> Result<()>;
+```
+1. What is expected?
+* TBD
+
+2. What DBS does?
+* Not implemented in DBS.
+
+---
+
+```rust
+    async fn resume_vm(&self) -> Result<()>;
+```
+1. What is expected?
+* This is a paired function with `pause_vm()`, it resumes the VMM's execution from the paused state.
+
+2. What DBS does?
+* Not implemented in DBS.
+
+---
+    
+```rust
+async fn add_device(&self, device: device::Device) -> Result<()>;
+```
+1. What is expected?
+* This functions is a generic interface of adding devices which implement trait `device::Device`. If the VMM
+is started, the device should be hotplugged. If the VMM have not started, the device should be in some buffer(pending_list) and plugged when VM starts. The function should only returns `Err` if the device is not supported or the 
```

Comment:
`buffer(pending_list)` -> `buffer(a pending devices list)`? `pending_list` looks like a variable name.

Reply:
`if the device is not supported or the` is not a complete sentence, looks like something has been truncated.

Reply:
Fixed.